elastic/elastic-agent
# [Fix] Agent incapable of running on Azure Container Instances

Bringing back #3139 

What this change is introducing on top of bringing back work introduced in #3084 is change of ordrer for some operations.

Changing owner of a file, discards capabilities set. This becomes a problem with heartbeat as it needs `setuid` and `netraw` capabilities to perform properly. 
So setting capabilities was moved after chown.

### How to test
- build docker image
- deploy to Azure Container Instances to see it running (it cannot fail)
- check capabilities `docker run --rm -it --entrypoint /bin/bash {image_id} -c "getcap data/elastic-agent-*/components/heartbeat"` -> this needs to have output (capabilities mentioned earlier)

I also checked with @justinkambic to see if we can test this with their failing E2E but it seems we need to merge it to see it fail. But it won't fail. OK?

Fixes: #82
